src/main/java/me/botsko/prism/serializers/items/ItemStackSerializer.java

```diff
@@ -0,0 +1,295 @@
+    public Map<String, String> bannerMeta;
+    public String material;
+
+    public static ItemStackSerializer createItemStackSerialized(ItemStack item) {
+        if (item == null || item.getAmount() <= 0) {
+            return null;
+        }
+        ItemStackSerializer data;
+        if (Tag.SHULKER_BOXES.isTagged(item.getType())) {
+            data = new StorageItemStackSerializer();
+            ItemMeta meta = item.getItemMeta();
+            if (meta instanceof BlockStateMeta) {
+                BlockState state = ((BlockStateMeta) meta).getBlockState();
+                if (state instanceof Container) {
+                    Inventory inv = ((Container) state).getInventory();
+                    inv.forEach(itemStack -> ((StorageItemStackSerializer) data).inventoryContents
+                            .add(createItemStackSerialized(itemStack)));
+                }
+            }
+        } else {
+            data = new ItemStackSerializer();
+        }
+        // Set basics
+        data.durability = (short) ItemUtils.getItemDamage(item);
+        data.amt = item.getAmount();
+        data.material = item.getType().name();
+        final ItemMeta meta = item.hasItemMeta() ? item.getItemMeta() : null;
+        if (meta != null) {
+            data.name = meta.getDisplayName();
+        }
+        if (meta instanceof LeatherArmorMeta) {
+            final LeatherArmorMeta lam = (LeatherArmorMeta) meta;
+            data.color = lam.getColor().asRGB();
+        } else if (meta instanceof SkullMeta) {
+            final SkullMeta skull = (SkullMeta) meta;
+            if (skull.hasOwner()) {
+                data.owner = Objects.requireNonNull(skull.getOwningPlayer()).getUniqueId().toString();
+            }
+        }
+
+        // Written books
+        if (meta instanceof BookMeta) {
+            final BookMeta bookMeta = (BookMeta) meta;
+            data.by = bookMeta.getAuthor();
+            data.title = bookMeta.getTitle();
+            data.content = bookMeta.getPages().toArray(new String[0]);
+        }
+
+        // Lore
+        if (meta != null && meta.hasLore()) {
+            data.lore = Objects.requireNonNull(meta.getLore()).toArray(new String[0]);
+        }
+
+        // Enchantments
+        if (!item.getEnchantments().isEmpty()) {
+            final String[] enchs = new String[item.getEnchantments().size()];
+            int i = 0;
+            for (final Map.Entry<Enchantment, Integer> ench : item.getEnchantments().entrySet()) {
+                // This is silly
+                enchs[i] = ench.getKey().getKey().getKey() + ":" + ench.getValue();
+                i++;
+            }
+            data.enchs = enchs;
+        } else if (meta instanceof EnchantmentStorageMeta) {
+            final EnchantmentStorageMeta bookEnchantments = (EnchantmentStorageMeta) meta;
+            if (bookEnchantments.hasStoredEnchants()) {
+                if (bookEnchantments.getStoredEnchants().size() > 0) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=6368169)